packages/lesswrong/lib/vulcan-users/permissions.ts

```diff
@@ -116,8 +116,11 @@ export const userOwns = function (user: UsersMinimumInfo|DbUser|null, document:
     return user._id === (document as HasUserIdType).userId;
   } else {
     // case 2: document is a user, use _id or slug to check
+    // TODO note fix for bug where admins can't edit their own slugs
     const documentUser = document as (DbUser|UsersMinimumInfo);
-    return documentUser.slug ? user.slug === documentUser.slug : user._id === documentUser._id;
+    const idsExistAndMatch = !!user._id && !!documentUser._id && user._id === documentUser._id;
+    const slugsExistAndMatch = !!user.slug && !!documentUser.slug && user.slug === documentUser.slug;
+    return idsExistAndMatch || slugsExistAndMatch;
```

Comment:
There was an edge case here where admins couldn't edit their own slugs, because it would prefer to use the slug rather than the id to decide if the user owned the object, so if the slug changed it would say the user didn't own their new user object. The thing that actually failed was some fields like `lastUsedTimezone` which have `userOwns` update permissions but not `'admins'`

Non-admins could edit their own slugs because we would use `documentId` in that case (now we always use slug), maybe that's why those two cases existed in the first place